test/debug/catch_test.rb

```diff
@@ -63,4 +63,47 @@ def test_debugger_stops_when_the_exception_raised
       end
     end
   end
+
+  class NamespacedExceptionCatchTest < TestCase
+    def program
+      <<~RUBY
+         1| class TestException < StandardError; end
+         2|
+         3| module Foo
+         4|   class TestException < StandardError; end
+         5|
+         6|   def self.raised_exception
+         7|     raise TestException
+         8|   end
+         9| end
+        10|
+        11| # we need this rescue + binding.bp workaround because the test framework can't handle exception exit yet
+        12| Foo.raised_exception rescue nil
+        13|
+        14| binding.bp
+      RUBY
+    end
+
+    def test_debugger_not_stopping_at_exception_without_namespace
+      debug_code(program) do
+        type 'catch TestException'
+        type 'continue'
+        assert_line_num(14)
+        type 'quit'
+        type 'y'
+      end
+    end
+
+    def test_debugger_stop_at_exception_with_namespace
```

Comment:
How about this one too?
```suggestion
    def test_catch_with_namespace_stops_at_exception
```